tests/rl/test_saoe_simple.py

```diff
@@ -0,0 +1,306 @@
+# Copyright (c) Microsoft Corporation.
```

Comment:
It raises error when running test with pandas-1.3.5

![image](https://user-images.githubusercontent.com/465606/167525863-c0d408c9-9e19-4b48-8bdb-bfd8cbbbcdd1.png)

Reply:
it raises less error when running test with pandas-1.4.2

![image](https://user-images.githubusercontent.com/465606/167525968-46d4d461-bd56-4645-9aa9-7caaa07acb78.png)

Reply:
numpy from 1.21.2 to  1.21.6 does not help
![image](https://user-images.githubusercontent.com/465606/167526248-fd7f531c-7c8a-4f59-838a-0fddde7aece8.png)

Reply:
install torch from 1.21.2  to 1.21.6 does not help
![image](https://user-images.githubusercontent.com/465606/167526765-74d891d6-b60d-44b5-a2a7-2e20618a6d53.png)

Reply:
@ultmaster   Do you have any suggestions about how to successfully run the test successfully locally? 

![image](https://user-images.githubusercontent.com/465606/167527144-6b13b0d4-6080-4be6-8e26-b9541fa364d8.png)

Reply:
That's strange. What's you tianshou version?

Could you post the full content of `.output/result.csv`?

Reply:
my `tianshou==0.4.8`:  
same as [the CI](https://github.com/microsoft/qlib/runs/6340717318?check_suite_focus=true#step:12:208)


Here is my output of  `pip freeze`
[env.txt](https://github.com/microsoft/qlib/files/8666117/env.txt)

Here is my `./tests/rl/.output/result.csv`
[result.csv](https://github.com/microsoft/qlib/files/8666145/result.csv)

@matluster

Reply:
Is this the only failed test? Or some other tests also failed?

Reply:
Could you print self.twap_price here, and see whether it equals to `market_price.mean()`?

<img width="667" alt="image" src="https://user-images.githubusercontent.com/79736092/167775393-71715c95-56c0-42df-bba3-60e6e5e07b87.png">

If not, try to see what's `self.ticks_for_oder` here.

<img width="740" alt="image" src="https://user-images.githubusercontent.com/79736092/167775459-b2a429fb-d8da-4b1a-844b-01674398accb.png">

Reply:
@ultmaster @matluster 
I'm trying to debug it.
First, I try to change it into a single process instead of multi-processing.
So, I made the following changes for DummyVectorEnv

![image](https://user-images.githubusercontent.com/465606/168479837-b6b46b5f-9bbf-4180-9f61-c0b1f1ec2cca.png)


But I got the following error when I ran it again.

Is it the correct way to debug the program? 

![image](https://user-images.githubusercontent.com/465606/168479962-5fb23b59-299b-46a2-afa2-fe77f25c513d.png)

Reply:
Nice catch. It was a flaw in this design. I added a hot-fix so that dummy env could work.

Now the tests have passed on my side. Please try again.

Reply:
@ultmaster  @matluster  
Thanks for your hot-fix

I try to debug.  Here is some information I got.

1)  I start the test by `pytest -s --pdb --disable-warnings /sdc/home/xiaoyang/re
  pos/qlib-main/libs/qlib/tests/rl/test_saoe_simple.py::test_twap_strategy`
It create 4 `DummyVectorEnv` and here is the first one.
It uses the deal price **all day** to get  twap_price.
![image](https://user-images.githubusercontent.com/465606/168768749-f65f85c4-7d76-4449-90b0-607861299f71.png)

2) I skip the next 3 initialization of `DummyVectorEnv` and stop at the first step in the first `DummyVectorEnv`
It uses the **first 30 tick market price** to get PA in the first step
![image](https://user-images.githubusercontent.com/465606/168769415-4cb6bf26-5f0e-45df-a6b1-b289cff4e5fe.png)

IIUC,  the PA at this step will not be zero (though the average PA all day will be 0).

But It tries to assert that the PA of **every step** is zero.  I don't think it is reasonable.
![image](https://user-images.githubusercontent.com/465606/168770580-0c2063f7-7f0b-4c11-baef-7ad6de706768.png)

Reply:
We didn't assert every PA at every **step** to be zero. Rather, we assert PA for every **episode** to be zero. There is a difference.

What you are referring to at L228 is the PA at every step. We actually never checked that (it's not even logged as a metric actually). What I'm interested in is the PA at L351.

I checked you result.csv, `market_price` and `trade_price` (at L351), which PA relies on, are both correct (and equal, because it's a TWAP strategy), but PA is wrong. One interesting fact is that the `market_price` here is actually "re-calculated", and not `self.twap_price`. I suggest something odd could happen here.

So my suggestion is to add the following lines at L351:

```python
print(self.twap_price, market_price.mean(), np.sum(market_price * exec_vol))
```

You would see three **equal** values on each line. If you didn't, something is wrong.

(But on the other hand, I see your `twap_price` is calculated on the whole-day price, which is actually... correct in this case. So I'm confused here.)

Reply:
@matluster @ultmaster  [This solution ](https://github.com/ultmaster/qlib/pull/2) will solve the problem in my system.
Please merge it